application/common/components/adapters/WorkdayIdStore.php

```diff
@@ -127,6 +130,8 @@ public function getAllActiveUsers()
             ), 1532982679);
         }
 
+        AdapterHelpers::addBlankProperty(self::ManagerEmail, $allActiveUsers);
```

Comment:
PHP 7.4 added the `??=` operator. [SO answer](https://stackoverflow.com/a/59102908/1461333)
```suggestion
        $allActiveUsers[self::ManagerEmail] ??= '';
```

Reply:
Do you think that's worth losing the value of having the FakeIdStore and the WordayIdStore call the same function? It's only the FakeIdStore code that gets tested.

Reply:
You can use the `??=` operator in the FakeIdStore as well.

Reply:
I don't think it matters that it's a nested array.

Reply:
You raise an interesting point, though. Since it's only testing against a fake ID store and not the real `idp-id-broker` it may not actually work.

Reply:
It doesn't look like it would work for the (inner) array elements.  It would just add a non-array entry with that key to the parent array.

Reply:
I don't know why it wouldn't. Your helper function doesn't recurse into nested arrays, so I don't see the difference.

Reply:
Doesn't this add the new entry to the child arrays rather than the parent array?

```
        foreach ($items as &$next) {
            if (!isset($next[$property])) {
                $next[$property] = '';
            }
        }
        
       ```

Reply:
Oh, I see the difference now. Sorry about that. If you want to use the `??=` operator it would be:
```
        foreach ($items as &$next) {
            $next[$property] ??= '';
        }
```

Reply:
If I do that I would also need to change the minimum php version in `composer.json` from `7.1` to `7.4`.  Is that worth it?

Reply:
Probably doesn't matter. It probably has less value for an app than it would have for a library. Either way is fine with me.

Reply:
Maybe I should actually bump it up to 8.1, since that is what the other Trello card is all about.